### PR TITLE
feat(filter items): implement feature filter items

### PR DESCRIPTION
**Pull request by HoangLiarPhuongLe**
- Create Categories Component
- Create Progress Bar Component
- Create Size Options Component
- Create Color Options Component
- Create Product Component
- Create List Products Component
- Create Slides Component
- Filter items by Categories, Colors and Sizes

image

### CODE REVIEW

File: practice-one/src/pages/Products/components/FilterOptions/Categories.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  value?: TCategory
+  setValue: React.Dispatch<React.SetStateAction<TCategory | undefined>>
+}) => {
+  return (
+    <div className="border-b-2 border-solid py-4">
+      <p className=" mb-2 text-lg capitalize">categories</p>
+      <div className="flex gap-2">
+        <Button
+          size="small"
+          checked={value === 'men'}
```
Comment (huyenpham-agilityio) [inline practice-one/src/pages/Products/components/FilterOptions/Categories.tsx:17]: create enum for these categories please

Comment (huyenpham-agilityio) [reply practice-one/src/pages/Products/components/FilterOptions/Categories.tsx:17]: update for size and color please

File: practice-one/src/pages/Products/components/Product/index.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      <div className="flex justify-between p-4 pb-6">
+        <div className="space-y-2">
+          <p className="font-bold">{name}</p>
+          <p className="text-slate-400">Best quality</p>
+          <div className="">
+            <ButtonIcon source={startfive} alt="start five" />
+            <ButtonIcon source={startfive} alt="start five" />
+            <ButtonIcon source={startfive} alt="start five" />
+            <ButtonIcon source={startfive} alt="start five" />
+            <ButtonIcon source={startfive} alt="start five" />
```
Comment (huyenpham-agilityio) [inline practice-one/src/pages/Products/components/Product/index.tsx]: duplicate code

File: practice-one/src/pages/Products/components/Product/index.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import product from '../../../../assets/images/product.png'
```
Comment (huyenpham-agilityio) [inline practice-one/src/pages/Products/components/Product/index.tsx]: productImage

File: practice-one/src/pages/Products/components/ProductsList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { TProduct } from '../../../../types'
+import Button from '../../../../components/Button'
+import Product from '../Product'
+import { Link } from 'react-router-dom'
+
+const ListProducts = ({ products }: { products: TProduct[] }) => {
+  return (
+    <div className="flex flex-col gap-5 ">
+      <div className="grid grid-cols-1 justify-between gap-x-8 gap-y-7  lg:grid-cols-3 ">
+        {products.map((product) => {
```
Comment (huyenpham-agilityio) [inline practice-one/src/pages/Products/components/ProductsList/index.tsx]: destructuring product

**Review by huyenpham-agilityio — CHANGES_REQUESTED** [review]

File: practice-one/src/pages/Products/components/FilterOptions/Categories.tsx (hunk 1 above)
Comment (HoangLiarPhuongLe) [reply practice-one/src/pages/Products/components/FilterOptions/Categories.tsx:17]: The type file is already in the branch `feature/implement-products-page` @huyenpham-agilityio 

image

**Pull request merged**